lib/collection/src/shards/local_shard.rs

```diff
@@ -492,12 +497,15 @@ impl LocalShard {
         // snapshot all shard's segment
         let snapshot_segments_shard_path = snapshot_shard_path.join("segments");
         create_dir_all(&snapshot_segments_shard_path).await?;
-        self.segments
-            .read()
-            .snapshot_all_segments(&snapshot_segments_shard_path)?;
 
-        // snapshot all shard's WAL
-        self.snapshot_wal(snapshot_shard_path)?;
+        {
```

Comment:
Maybe this whole block should be running within `tokio::task::spawn_blocking` as it performs a lot of disk operations on a regular tokio thread.